src/mixins/README.md

```diff
@@ -0,0 +1,136 @@
+# Mixins
+
+This directory contains source files for the mixins exported by this extension.
+
+## Overview
+
+Sometimes it's helpful to write partial classes and then combining them together to build more powerful classes. This pattern is called Mixins (mixing in partial classes) and is supported by LoopBack 4. 
+
+LoopBack 4 supports mixins at an `Application` level. Your partial class can then be mixed into the `Application` class. A mixin class can modify or override existing methods of the class or add new ones! It is also possible to mixin multiple classes together as needed.
+
+### High level example
+```ts
+class MyApplication extends MyMixinClass(Application) {
+  // Your code
+};
+
+// Multiple Classes mixed together
+class MyApp extends MyMixinClass(MyMixinClass2(Application)) {
+  // Your code
+}
+```
+
+## Getting Started
+
+For hello-extensions we write a simple Mixin that allows the `Application` class to bind a `Logger` class from ApplicationOptions, Components, or `.logger()` method that is mixed in. `Logger` instances are bound to the key `loggers.${Logger.name}`. Once a Logger has been bound, the user can retrieve it by using [Dependency Injection](http://loopback.io/doc/en/lb4/Dependency-injection.html) and the key for the `Logger`.
+
+### What is a Logger?
+> A Logger class is provides a mechanism for logging messages of varying priority by providing an implementation for `Logger.info()` & `Logger.error()`. An example of a Logger is `console` which has `console.log()` and `console.error()`. 
+
+#### An example Logger
+```ts
+class ColorLogger implements Logger {
+  log(...args: LogArgs) {
+    console.log('log  :', ...args);
+  }
+
+  error(...args: LogArgs) {
+    const data = args.join(' ');
+    // log in red color
+    console.log('\x1b[31m error: ' + data + '\x1b[0m');
```

Comment:
Maybe `console.log('\x1b[31m error:', ...args, '\x1b[0m')` for consistency with the `log` method? `args.join(' ')` will convert object values to string via `.toString()`, `console.log` uses `util.format` instead.